File: users/allTests/e2eTests/userAddForm.py
# -*- coding: utf-8 -*-

from django.contrib.staticfiles.testing import StaticLiveServerTestCase
from subprocess import call

from users.models import SysUser
from .userLogin import create_test_user


class UserAddFormTest(StaticLiveServerTestCase):
    """All tests with nightwatch and Selenium server"""

    def test_userAddForm(self):
        create_test_user()
        # calling browser
        self.assertEqual(0,
                         call('nightwatch --test users/allTests/e2eTests/userAddForm.js', shell=True),
                         'Nighwatch tests failed')
        user = SysUser.objects.last()
        self.assertEqual(user.username, 'username')
        self.assertEqual(user.first_name, 'First name')
        self.assertEqual(user.last_name, 'Last name')
        self.assertEqual(user.rank, 'kpr.')
        self.assertEqual(user.email, 'user@user.com')

File: users/allTests/e2eTests/userAddForm.js
module.exports = {
    'login to system': function (browser) {
        browser
            .login('test_user', 'test_user')
    },
    'testing user add form': function (browser) {
        browser
            .url('http://127.0.0.1:8081/#/users')
            .waitForElementVisible('button[ng-click="openUserAddModal()"]', 2000)
            .click('button[ng-click="openUserAddModal()"]')
            .waitForElementVisible('form[name=userAddForm]', 2000)
            .setValue('input[id="id_username"]', 'username')
            .setValue('input[id="id_first_name"]', 'First name')
            .setValue('input[id="id_last_name"]', 'Last name')
            .setValue('select[id="id_rank"]', ['kapral', browser.Keys.RETURN])
            .setValue('input[id="id_email"]', 'user@user.com')
            .click('button[ng-click="ok()"]')
            .pause(500)
            .end()
    }
};